src/main/java/com/dataguise/saas/service/DetectionTaskService.java

```diff
@@ -0,0 +1,192 @@
+     * @throws BadGatewayException
+     * @throws BadRequestException
+     */
+    public Integer editTask(@PathVariable Integer id, @RequestBody DgDiscoverTask detectionTask)
+            throws BadGatewayException, BadRequestException {
+        Integer taskId = null;
+        if( id == null) {
+            throw new BadRequestException("Task " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage());
+        }
+        try {
+            taskId = restDriver.editDetectionTask(id, detectionTask);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Detection task edited with id: " + taskId);
+        return taskId;
+    }
+
+    /**
+     * Execute a detection task
+     * @param taskId
+     * @return
+     * @throws BadRequestException
+     * @throws BadGatewayException
+     */
+    public Integer executeTask(@PathVariable Integer taskId) throws BadRequestException, BadGatewayException {
+        if( taskId == null) {
+            throw new BadRequestException("Task " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage());
+        }
+        Integer taskInstanceId = null;
+        try {
+            taskInstanceId = restDriver.executeDetectionTask(taskId);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Detection task executed with instance id: " + taskInstanceId);
+        return taskInstanceId;
+    }
+
+    /**
+     * Construct the DgDiscoverTask to create a detection task
+     * @param connectionIdList
+     * @return
+     * @throws BadGatewayException
+     * @throws InternalServerException
+     */
+    public DgDiscoverTask constructDetectionTask(List<Integer> connectionIdList) throws BadGatewayException,InternalServerException {
+        DgDiscoverTask discoverTask = null;
+        Integer taskInstanceId = null;
+        try {
+            String taskDetail = taskBuilderService.constructTaskName(taskBuilderService.DETECTION_TYPE);
+
+            DetectionTask.DetectionTaskBuilder builder = new DetectionTask.DetectionTaskBuilder(taskDetail, taskDetail, Modules.RDS)
+                    .setTaskType(DetectionTask.TaskTypes.Discover)
+                    .setSearchType("Deep")
+                    .useExistingSafeList(false)
+                    .setSampleDataStart(DetectionTask.SampleDataStart.Top)
+                    .setSampleSizeSpecifcation(DetectionTask.SampleSizeSpecification.Rows, 1000)
+                    .setAdditionalSampleSize(0.0)
+                    .showSampleData(true)
+                    .showViews(true);
+
+            builder = taskBuilderService.addTaskBuilderPolicies(builder);
+
+            for (Integer connectionId : connectionIdList) {
+                builder.addConnection(connectionId);
+            }
+            logger.debug("Detection task builder bean to driver: "+builder);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/DetectionTaskService.java L124
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-971133967>*